app/helpers/sprangular/application_helper.rb

```diff
@@ -22,19 +22,44 @@ def js_environment
         end
       ]
 
-      {env: Rails.env,
+      {
+        env: Rails.env,
         config: {
           site_name: store.seo_title || store.name,
           logo:      asset_path(config.logo),
+          locale:    I18n.locale,
+          currency:  Money::Currency.table[current_currency.downcase.to_sym],
+          supported_locales: supported_locales,
           default_country_id: config.default_country_id,
           payment_methods: payment_methods,
           image_sizes: Spree::Image.attachment_definitions[:attachment][:styles].keys,
           product_page_size: Spree::Config.products_per_page
         },
+        translations: current_sprangular_translations,
         templates: templates
       }
     end
 
+    def supported_locales
+      if Object.const_defined?('SpreeI18n')
+        SpreeI18n::Config.supported_locales
+      else
+        # Use Default locale, and do not provide a front end selector.
+        [:en, :de]
+      end
+    end
+
+    ##
+    # Get relevant translations for front end. For both a simple, and "Chainable"
+    # i18n Backend, which is used by spree i18n.
+    def current_sprangular_translations
```

Comment:
Assignment Branch Condition size for current_sprangular_translations is too high. [15.56/15]